coalib/misc/CachingUtilities.py

```diff
@@ -0,0 +1,174 @@
+    is corrupted and will be removed.
+
+    :param log_printer: A LogPrinter object to use for logging.
+    :param files:       The list of files to be deleted.
+    :return:            True if all the given files were successfully deleted.
+                        False otherwise.
+    """
+    error_files = []
+    for file_name in files:
+        file_path = get_cache_data_path(log_printer, file_name)
+        cache_dir = os.path.dirname(file_path)
+        try:
+            os.remove(file_path)
+        except OSError:
+            error_files.append(file_name)
+
+    if len(error_files) > 0:
+        error_files = ", ".join(error_files)
+        log_printer.warn("There was a problem deleting the following "
+                         "files: " + error_files + ". Please delete "
+                         "them manually from " + cache_dir)
+        return False
+
+    return True
+
+
+def pickle_load(log_printer, filename, fallback=None):
+    """
+    Get the data stored in ``filename`` present in the user
+    config directory. Example usage:
+
+    >>> from pyprint.NullPrinter import NullPrinter
+    >>> from coalib.output.printers.LogPrinter import LogPrinter
+    >>> log_printer = LogPrinter(NullPrinter())
+    >>> test_data = {"answer": 42}
+    >>> pickle_dump(log_printer, "test_file", test_data)
+    >>> pickle_load(log_printer, "test_file")
+    {'answer': 42}
+    >>> pickle_load(log_printer, "nonexistant_file")
+    >>> pickle_load(log_printer, "nonexistant_file", fallback=42)
+    42
+
+
+    :param log_printer: A LogPrinter object to use for logging.
+    :param filename:    The name of the file present in the user config
+                        directory.
+    :param fallback:    Return value to fallback to in case the file doesn't
+                        exist.
+    :return:            Data that is present in the file, if the file exists.
+                        Otherwise the ``default`` value is returned.
+    """
+    filename = get_cache_data_path(log_printer, filename)
+    if not os.path.isfile(filename):
+        return fallback
+    with open(filename, "rb") as f:
+        try:
+            return pickle.load(f)
+        except (pickle.UnpicklingError, EOFError) as e:
+            log_printer.warn("The caching database is corrupted and will "
+                             "be removed. Each project will be re-cached "
+                             "automatically in the next run time.")
+            delete_cache_files(log_printer, files=[filename])
+            return fallback
+
+
+def pickle_dump(log_printer, filename, data):
+    """
+    Write ``data`` into the file ``filename`` present in the user
+    config directory.
+
+    :param log_printer: A LogPrinter object to use for logging.
+    :param filename:    The name of the file present in the user config
+                        directory.
+    :param data:        Data to be serialized and written to the file using
+                        pickle.
```

Comment:
Actually there is no return value (`None`). Dunno why I put that (not using it anyway). Removed `return`.